Assets/Scripts/Player/Body/Body.cs

```diff
@@ -101,5 +110,30 @@ private BodyPart FindBodyPart(BodyPartType bodyPartType)
 		{
 			return bodyParts.FirstOrDefault(bodyParts => bodyParts.BodyPartType == bodyPartType);
 		}
-	}
+
+        private void UpdateBodyPartList()
+        {
+            bodyParts.Clear();
+            foreach (BodyPart part in gameObject.GetComponentsInChildren<BodyPart>())
+            {
+                bodyParts.Add(part);
+            }  
+        }
+
+        private void UpdateBodyPartVisuals(GameObject newBodyPart, BodyPart bodyPart)
+        {
+            // Making sure the part is equal to the original one
+            SkinnedMeshRenderer newBodyPartMesh = newBodyPart.GetComponent<SkinnedMeshRenderer>();
+            SkinnedMeshRenderer bodyPartMesh = bodyPart.SkinnedMeshRenderer;
+
+            Material[] materials = bodyPartMesh.sharedMaterials;
+            newBodyPartMesh.materials = materials;
+
+            // there's 5 shape keys on the human model
+            for (int i = 0; i < 4; i++)
```

Comment:
Just don't hardcode the number of blendshapes

Reply:
Just saw the comment above. The answer is `for (int i = 0; i < newBodyPartMesh.sharedMesh.blendShapeCount; i++)`

Reply:
wasn't aware of that. thanks.